### ISSUE TITLE
MeshVisual flat and smooth shading broken for 2D meshes

### ISSUE THREAD

**Issue by djhoese** [vispy/vispy#1861]
I just tried this out, it works fine for me all my use cases - including after dropping the lines @djhoese flagged as not needed (which should be done before this gets merged).

I did notice a problem with `flat` and `smooth` shading when using a 2D mesh - which is the hardcoding of some some 3D assumptions here https://github.com/vispy/vispy/blob/9e0822db0ec31ba720d82952b1867e77f30971f6/vispy/geometry/meshdata.py#L339 and here https://github.com/vispy/vispy/blob/9e0822db0ec31ba720d82952b1867e77f30971f6/vispy/geometry/meshdata.py#L307
but that is really independent of this PR and would have been broken before hand. When the time is right I can make a short follow-up PR to make sure the those lines do something reasonable when given 2D meshes if a shading mode is enabled

_Originally posted by @sofroniewn in https://github.com/vispy/vispy/pull/1462#issuecomment-570293597_
